<div style="border:solid Purple 2px; padding: 40px">

<b>Привет, Максим!👋

Меня зовут Эльвира, я буду ревьюером твоего проекта. Предлагаю общаться на «ты», но если это не удобно - дай мне знать, и мы перейдем на «вы».


Ты можешь найти мои комментарии, обозначенные <font color='green'>зеленым</font>, <font color='gold'>желтым</font> и <font color='red'>красным</font> цветами, например:


<div class="alert alert-success">
<h2> Комментарий ревьюера 😊<a class="tocSkip"> </h2>
    
Такими комментариями я буду помечать отлично проделаную работу😉
</div>
    
<div class="alert alert-warning">
<h2> Комментарий ревьюера 🤓<a class="tocSkip"> </h2>
        
В таких комментариях я постараюсь подсказать тебе более элегантное или легкое решение, некоторые хитрости и фишки. Части проекта, помеченные такими комментариями, можно не исправлять, но рекомендую обратить на них внимание.</div>

<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера ⛔️<a class="tocSkip"></h2>
    
В случае, когда решение на отдельном шаге требует существенной переработки и внесения правок. Если ты видишь такой комментарий, значит здесть есть недочет, который следует исправить.
</div>

Ты также можешь реагировать на мои комментарии своими, выделяя их цветами и наиболее понравившимся тебе способом оформления, но явно  отличающимся от моих комментариев. Это нужно, чтобы не создавалась путаница🙃
    
`<div class="alert alert-info"> <b>Комментарий студента:</b> Например, вот так.</div>`

Чтобы сделать подобный блок, кликни здесь дважды и скопируй всю предыдущую строку ;)
    
Пожалуйста, не удаляй и не перемещай мои комментарии, они будут особенно полезны для нашей работы в случае повторной проверки проекта.</div></b>

<div class="alert alert-info"> 
<b>Комментарий студента:</b> 
    
Привет. Отлично на 'Ты'. 

</div>

# Исследование надежности заемщиков


<div class="alert alert-warning">
<h2> Комментарий ревьюера 🤓<a class="tocSkip"> </h2>

Введение, как и общий вывод - важные части твоей работы, на которые в первую очередь обращает внимание заказчик. Введение станет лучше, если ты добавишь в него:
* заголовок,

* краткое описание проекта,

* выделим цель проекта

* описание данных

* краткий план работы, выделим основные этапы, которые должны сделать.

Тогда получатель отчета сможет быстрей его понять.

</div>

<div class="alert alert-info"> 
<b>Комментарий студента:</b> 
    
Спасибо буду уделять этому внимание.

</div>

Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

#data = pd.read_csv('data.csv')

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

<div class="alert alert-warning">
<h2> Комментарий ревьюера 🤓<a class="tocSkip"> </h2>
Кажется, работа над проектов велась локально и пути к файлам не были изменены на общедоступные. Сейчас я внесу изменения в твой код, но в следующий раз будь внимательней. Стоит писать код таким образом, что бы он запускался и в jupiter hub, и локально без внесения дополнительных правок. Этого можно добиться многими способами. Например, использовать конструкцию ``try-except`` — для путей файлов: ``try`` — пути на локальном компьютере, ``except``  пути на сервере. Или применить библиотеку ``os``:

    import os

    pth1 = '/folder_1/data.csv'
    pth2 = '/folder_2/data.csv'
    
    if os.path.exists(pth1):
        query_1 = pd.read_csv(pth1)
    elif os.path.exists(pth2):
        query_1 = pd.read_csv(pth2)
    else:
        print('Something is wrong')

<div class="alert alert-info"> <b>Комментарий студента:</b> Правильное замечание)))</div>

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
days_employed_nan = data['days_employed'].isna().sum() # переменная с количеством пропусков столбца Трудовой стаж
total_income_nan = data['total_income'].isna().sum() # переменная с количеством пропусков столбца ежемесячный доход


data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:

#print(data.groupby('children')['debt'].sum() / data.groupby('children')['children'].count() * 100) 

# выведем на экран соотношение в процентах по группам имеющих задолженность с общим количеством клиентов без детей и сдетьмы и их количеством

child_debt = data.groupby('children').agg({'debt':['count','sum']}) #группировка списка по количеству детей и подсчет числа в группе и суммы дебиторов
child_debt.columns = ['total','debt']
child_debt['%'] = child_debt['debt'] / child_debt['total'] * 100
child_debt



# Ваш код будет здесь. Вы можете создавать новые ячейки.



total  debt         %
children                       
0         14091  1063  7.543822
1          4808   444  9.234609
2          2052   194  9.454191
3           330    27  8.181818
4            41     4  9.756098
5             9     0  0.000000

<div class="alert alert-warning">
<h2> Комментарий ревьюера 🤓 v_2 <a class="tocSkip"> </h2>

Столбец '%' можно еще рассчитать как среднее (`mean`), это и будет наша доля должников по каждой категории.

In [27]:
print(data.groupby('children').agg({'debt':['count','sum','mean']}))

           debt                
          count   sum      mean
children                       
0         14091  1063  0.075438
1          4808   444  0.092346
2          2052   194  0.094542
3           330    27  0.081818
4            41     4  0.097561
5             9     0  0.000000


<div class="alert alert-info"> <b>Комментарий студента:</b> точно, сработало и все в одну строчку</div>


<div class="alert alert-warning">
<h2> Комментарий ревьюера 🤓 v_2 <a class="tocSkip"> </h2>

Только 'count' тоже следует оставить)

**Вывод:** 

    Получил данные о 21331 клиенте, из которых 14091 не имеют детей и 7240 с детьми.

    Из них:
    - 4808 - с одним ребенком,
    - 2052 - двое детей,
    - 330 - трое детей,
    - 41 - четверо детей,
    - 9 - имеют пятерых детей.

    Получил долю плательщиков имеющих задержками по платежам, среди этих категорий   
   
    - 7.54% - без детей
    - 9,23% - с одним ребенком,
    - 9,45% - двое детей,
    - 8,18% - трое детей,
    - 9,75% - четверо детей,
    - 0% - имеют пятерых детей.
    
    Могу утверждать что зависимость между количеством детей и возвратом кредита в срок не наблюдается. 
    Видно полностью вовремя погашенную задолженность у клиентов с пятью детьми, но данных слишком мало для получения полной картины.
    Важно отметить клиенты без детей (7,54%) не смотря на более высокую долю присутствия имеют лучшую статистику по возвратам кредита вовремя.
    

<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера ⛔️<a class="tocSkip"></h2>
<s>
В данном пункте кол-ва заемщиков по категориям с 3-мя и более детьми недостаточно для достоверных выводов. А если рассмотреть самые многочисленные выборки с 0, 1 и 2-мя детьми - можно увидеть, что чем больше у заемщика детей, тем он хуже платит в срок.

<div class="alert alert-info"> <b>Комментарий студента:</b> 

Все верно! Главное, в случае с уровнем дохода обратил внимание на выработку а тут нет.

**Вывод2:** 

    Получил данные о 21331 клиенте, из которых 14091 не имеют детей и 7240 с детьми.

    Из них:
    - 4808 - с одним ребенком,
    - 2052 - двое детей,
    - 330 - трое детей,
    - 41 - четверо детей,
    - 9 - имеют пятерых детей.

    Получил долю плательщиков имеющих задержками по платежам, среди этих категорий   
   
    - 7.54% - без детей
    - 9,23% - с одним ребенком,
    - 9,45% - двое детей,
    - 8,18% - трое детей,
    - 9,75% - четверо детей,
    - 0% - имеют пятерых детей.
    
    Могу утверждать, что зависимость между количеством детей и возвратом кредита в срок наблюдается у клиентов категорий с 1 и 2-мя детьми, с появлением второго ребенка % риска увеличивается. 
    Видно полностью вовремя погашенную задолженность у клиентов с пятью детьми, но данных слишком мало для получения полной картины, так же как и в группах с тремя и четырьмя детьми.
    Важно отметить клиенты без детей (7,54%) не смотря на более высокую долю присутствия имеют лучшую статистику по возвратам кредита вовремя.
    
</div>


<div class="alert alert-success">
<h2> Комментарий ревьюера 😊 v_2 <a class="tocSkip"> </h2>

Теперь все отлично) 

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [ ]:


#print(data.groupby('family_status')['debt'].sum() / data.groupby('family_status')['family_status'].count() * 100)
# выведем на экран соотношение в процентах по группам имеющих задолженность /
#с общим количеством клиентов сгруппировав по семейному положению('family_status')

family_status_debt = data.groupby('family_status').agg({'debt':['count','sum']}) #группировка списка по семейному статусу и подсчет числа в группе и суммы дебиторов
family_status_debt.columns = ['total','debt']
family_status_debt['%'] = family_status_debt['debt'] / family_status_debt['total'] * 100
family_status_debt



# Ваш код будет здесь. Вы можете создавать новые ячейки.

**Вывод:** 
    
    По семейному положению клиенты делятся и имеют следующий процент не возврата кредита в срок:
| Семейное положение        | Количество клиентов | Доля возвратов не вовремя |
|---------------------------|---------------------|----------------------------|
| женат / замужем           |12261                |7,56%                       |
| гражданский брак          |4134                 |9,31%                       |
| Не женат / не замужем     |2796                 |9,76%                       |
| в разводе                 |1189                 |7,06%                       |
| вдовец / вдова            |951                  |6,62%                       |

Самая высокая доля не вовремя выплаченных кредитов приходится на `Не женат / не замужем(9,76%)`. И наоборот уменьшается при образовании совместной жизни `гражданский брак(9,31%)` потом семьи `женат / замужем(7,56%)` и после брака в других категориях тоже доля надежных клиентов выше`(до 6,62%)`. Зависимость между семейным положением и возвратом кредита в срок есть.


<div class="alert alert-warning">
<h2> Комментарий ревьюера 🤓<a class="tocSkip"> </h2>

Можно отметить общую тенденцию, что люди не состоящие в оф. браке склонны к задержкам)

<div class="alert alert-info"> <b>Комментарий студента:</b> Согласен если эти люди не были ранее в браке)))</div>


<div class="alert alert-success">
<h2> Комментарий ревьюера 😊 v_2 <a class="tocSkip"> </h2>

Да, не состоящие и не состоявшие)

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [ ]:


#print(data.groupby('total_income_category')['debt'].sum() / data.groupby('total_income_category')['total_income_category'].count() * 100)
# выведем на экран соотношение в процентах по группам имеющих задолженность /
#с общим количеством клиентов сгруппировав по уровню дохода('total_income_category')

total_income_category_debt = data.groupby('total_income_category').agg({'debt':['count','sum']}) #группировка списка по уровню дохода и подсчет числа в группе и суммы дебиторов
total_income_category_debt.columns = ['total','debt']
total_income_category_debt['%'] = total_income_category_debt['debt'] / total_income_category_debt['total'] * 100
total_income_category_debt



# Ваш код будет здесь. Вы можете создавать новые ячейки.

**Вывод:** 

    По уровню дохода клиенты делятся и имеют следующий процент не возврата кредита в срок:
| Уровень дохода | Количество клиентов | Доля возвратов не вовремя |
|-|-|-|
| А |25 |8,00% |
| B |5014 |7,06%|
| C |15921|8,49%|
| D |349|6,01%|
| E |22|9,09%|

Зависимость между уровнем дохода и возвратом кредита в срок могу с уверенность отследить по группам с доходом категорий `С` и `В`, так как данных хватит для объективной оценки. Видно, что с увеличением заработка доля по невыплаченным кредитам вовремя уменьшилась на 1,5 % пункта. Если не учитывать малое количество клиентов категорий А D и Е то самые надежные конечно D(6.01%)

    

<div class="alert alert-success">
<h2> Комментарий ревьюера 😊<a class="tocSkip"> </h2>

Верно, молодец)

<div class="alert alert-info"> <b>Комментарий студента:</b> Спасибо, значит не обязательно разбивать на категории по меньшим суммам дохода, а то я уже приготовился</div>


<div class="alert alert-success">
<h2> Комментарий ревьюера 😊 v_2 <a class="tocSkip"> </h2>

Можно дополнить еще одной категоризацией, ее можно будет предложить заказчику)

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [ ]:


#print(data.groupby('purpose_category')['debt'].sum() / data.groupby('purpose_category')['purpose_category'].count() * 100)
# выведем на экран соотношение в процентах по группам имеющих задолженность /
#с общим количеством клиентов сгруппировав по цели кредита('purpose_category')

purpose_category_debt = data.groupby('purpose_category').agg({'debt':['count','sum']}) #группировка списка по цели кредита и подсчет числа в группе и суммы дебиторов
purpose_category_debt.columns = ['total','debt']
purpose_category_debt['%'] = purpose_category_debt['debt'] / purpose_category_debt['total'] * 100
purpose_category_debt



# Ваш код будет здесь. Вы можете создавать новые ячейки.

**Вывод:** 

    По целям на кредит клиенты делятся и имеют следующий процент не возврата кредита в срок:
| Цель кредита | Количество клиентов | Доля возвратов не вовремя |
|-|-|-|
| операции с автомобилем |4279 |9,35% |
| операции с недвижимостью |10751 |7,26%|
| получение образования |3988|9,25%|
| проведение свадьбы |2313|7,91%|

Самая низкая доля по возвратам не вовремя `7,26%` у `операции с недвижимостью` и самая высокая `9,35%` у `операции с автомобилем`



<div class="alert alert-success">
<h2> Комментарий ревьюера 😊<a class="tocSkip"> </h2>

Верно)

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

In [ ]:

rows_number = data.shape[0] # количество строк
print(f'Доля пропусков трудового стажа: {(days_employed_nan/rows_number):.0%}') # выводим на экран какую долю составляют пропущенные значения в слобце трудовой стаж
print(f'Доля пропусков ежемесячных доходов: {(total_income_nan/rows_number):.0%}') # выводим на экран какую долю составляют пропущенные значения в слобце ежемесячный доход

*Ответ:* Доли пропусков данных высокие 10% от общего числа. Пропуски не случайны по происхождению, возможно чтобы скрыть отсутствие или небольшой стаж работы, клиент или даже кредитный специалист не заполнял данные. Доход могли скрывать по этим же причинам либо наоборот не хотят его разглашать.

<div class="alert alert-warning">
<h2> Комментарий ревьюера 🤓<a class="tocSkip"> </h2>

Возможен и технический сбой)

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 
    
   При обработке большого массива данных, где пропуска в ячейках с долей в 10% от общего количества строк, может привести к неверной выборке и утерей ценных данных, так как не получится использовать информацию других столбцов этих строк.
    Во избежание этого необходимо максимально заполнить пропуски.
    В случае количественных переменных лучшее решение заполнить пропуски медианным значением. 
    Такой метод позволит исключить влияние всплесковых или аномальных данных. К тому же при  более детальной группировке свойственной влиять на данный показатель (например должность, образование и т.п.) мы можем приблизить подставляемое значение к естественному.


<div class="alert alert-success">
<h2> Комментарий ревьюера 😊<a class="tocSkip"> </h2>

Точно, медиана менее чувствительна к выбросам, чем среднее.

### Шаг 4: общий вывод.


Проверил 4 гипотезы влияния на возврат кредита в срок.
1. Есть ли зависимость между количеством детей и возвратом кредита в срок
Количество детей в семье при увеличении или уменьшении не влияет на выплату кредита, возможно с исследованием более крупного массива данных можно будет сказать как коррелируют эти показатели.
При этом влияние наличия детей в семье да однозначно влияет и если сравнивать с заемщиками без детей.

2. Есть ли зависимость между семейным положением и возвратом кредита в срок.
Да зависимость есть. Надежность клиентов повышается по данному жизненному пути. Так до брака риск с клиентом на уровне 9,76% а после него снижается до 6,62%

3. Есть ли зависимость между уровнем дохода и возвратом кредита в срок.
Зависимость влияния дохода исходя из выработки получилось отследить на двум категориям и она положительная(-1,43%). При увеличении выборки, а так же категоризации на более частые группы по доходу представится возможность отследить динамику.

4. Как разные цели кредита влияют на его возврат в срок.
При анализе целей на кредит выявил самый надежный сегмент это ’операции с недвижимостью’  - риск 7,26% так же эта доля подкрепляется большим процентом в общей массе выданных займов 50%. 
Фактор цели влияет и его рейтинг от недвижимости к проведению свадьбы (7,91%), получению образования(9,25%) и операциям на автомобиль(9,35%), обусловлен так же другими факторами возраст, осознанность принятого решения, кредитная история на момент заключения операций с недвижимостью.

Важно обратить внимание на пропуски данных, которые важно понимать и не допускать пропуска. Подтвердилась гипотеза о влиянии дохода на возврат кредита. Необходимы меры, например не пропускать заявку на согласование без указания этих данных с проверкой на программном уровне и специалистов кредитного отдела.



<div class="alert alert-success">
<h2> Комментарий ревьюера 😊<a class="tocSkip"> </h2>

Содержательный вывод по проделанной работе)

<div style="border:solid Purple 2px; padding: 40px">

<h2> Общий комментарий ревьюера 😊<a class="tocSkip"> </h2>
    <br/>
Ты проделал большую работу, и все основные задачи в рамках проекта решены, это отличная работа! Особенно хочу выделить  подробные выводы и работу со сводными таблицами!)
 
    
 
**Желаю удачи и жду твой проект на повторное ревью! Если вдруг у тебя возникнут вопросы, то я с радостью отвечу на них.    😊**
    

<div class="alert alert-info"> <b>Комментарий студента:</b> 

Благодарю, за обратную связь. Приятно научится новому и сейчас.

Кстати как ты вставляешь смайл?
    
Где порекомендуешь нарабатывать практику дополнительно? Я сейчас понимаю, что если мне каждый день не писать, код буду терять приобретенные навыки.

</div>


<div style="border:solid Purple 2px; padding: 40px">

<h2> Общий комментарий ревьюера 😊 v_2<a class="tocSkip"> </h2>
    <br/>
Максим, спасибо за проект! Успехов в дальнейшем обучении!)
    
Смайлы через ctrl+c и ctrl+v. Лучше всего в доп. проектах. Можно после сдачи работать над теми же проектами, проводить дополнительные анализы и пробовать новые фишки) 
    